### PR TITLE
Able to insert table columns

### PR DESCRIPTION
**Pull request by uzxmx**
This PR is mainly about inserting new columns, and can solve #119 , #130.

Also, I've tweaked logic of deleting columns slightly. Before the changes, when cursor is not in the column header, the deleting column command will not delete the whole column, which is caused by calling MoveToFirstRow rather than moving to the very first row(which should be the header or the border or the first row).

All tests passed on my local machine.

### CODE REVIEW

File: autoload/tablemode/table.vim [hunk 1]
```diff
@@ -163,7 +163,7 @@ function! tablemode#table#IsHeader(line) "{{{2
 endfunction
 
 function! tablemode#table#IsRow(line) "{{{2
-  return !tablemode#table#IsBorder(a:line) && getline(a:line) =~# (tablemode#table#StartExpr() . g:table_mode_separator) . '[^' . g:table_mode_separator . ']\+'
+  return !tablemode#table#IsBorder(a:line) && getline(a:line) =~# (tablemode#table#StartExpr() . g:table_mode_separator . '\+') . '[^' . g:table_mode_separator . ']\+'
```
Comment (uzxmx) [inline autoload/tablemode/table.vim]: I changed this in order to have line like `|| foo | bar |` or `||| foo | bar |` to be identified as a table row, so when inserting columns at the first column, rows can be identified as valid table row to do realignment.

**Comment by uzxmx** [comment]
@dhruvasagar

**Comment by dhruvasagar** [comment]
@uzxmx Thanks a lot! I am happy you included tests too, will test this out and merge soon.

File: autoload/tablemode/spreadsheet.vim [hunk 2]
```diff
@@ -152,6 +170,25 @@ function! tablemode#spreadsheet#DeleteColumn() "{{{2
...
+      call tablemode#spreadsheet#MoveToHeaderOrBorderOrFirstRow()
+      if a:insertAfter
+        call tablemode#spreadsheet#MoveToEndOfCell()
+        silent! execute "normal! 2l\<C-V>"
+      else
+        call tablemode#spreadsheet#MoveToStartOfCell()
+        silent! execute "normal! h\<C-V>"
+      endif
+      call tablemode#spreadsheet#MoveToLastRow()
+      normal! I|
```
Comment (dhruvasagar) [inline autoload/tablemode/spreadsheet.vim]: This needs to be `exec 'normal! I' . g:table_mode_separator`

Comment (uzxmx) [reply autoload/tablemode/spreadsheet.vim]: 👍  I will update.

**Comment by uzxmx** [comment]
@dhruvasagar  I don't get what you meant by "breaks the existing borders if we have any, and they need to be readded manually again", could you give me an example?

**Comment by uzxmx** [comment]
Hi @dhruvasagar, I've solved the issue that I knew. Is there any issue that needs to be solved?

**Comment by dhruvasagar** [comment]
@uzxmx Hi, yea there's a small usability issue I described above, when you add a column to a table that has a border separating the headers, the headers break after adding the column. I tried fixing it but haven't found enough time yet. I will take another crack at it tomorrow. Will let you know, thanks for your efforts!

**Pull request closed without merging**